scripts/src/public/middleware.ts

```diff
@@ -3,13 +3,23 @@ import * as bearerToken from "express-bearer-token";
 import * as httpContext from "express-http-context";
 
 import { logRequest, reportMetrics, requestLogger } from "../middleware";
+import { BlockchainEndpointChanged } from "../errors";
 
-export * from "../middleware";
+export { notFoundHandler, generalErrorHandler, statusHandler } from "../middleware";
+
+const deprecationError = function(req: express.Request, res: express.Response, next: express.NextFunction) {
+	const shouldRaiseError = req.header("x-simulate-deprecation-error");
+	if (shouldRaiseError) {
+		throw BlockchainEndpointChanged("simulated deprecation");
+	}
+	next();
+} as express.RequestHandler;
 
 export function init(app: express.Express) {
 	app.use(httpContext.middleware as express.RequestHandler);
 	app.use(bearerToken());
 	app.use(requestLogger);
 	app.use(logRequest);
 	app.use(reportMetrics);
+	app.use(deprecationError);
```

Comment:
will there be an overhead of checking every request for this specific header?

Reply:
It's just another if, and I can also remove it on production for now. The client just needs a way to test the error functionality

Reply:
this is weird. not that we should care about the extra function for each request, it's just that this is needed for a very specific thing only on dev/test.
why can't we just open a specific endpoint for it?